backend/windmill-queue/src/jobs.rs

```diff
@@ -684,9 +690,113 @@ pub async fn pull<R: rsmq_async::RsmqConnection + Clone>(
         }
     }
 
+    let final_workspaces_filter = workspaces_filter.as_str();
+
     // let rs = rd_string(2);
     // let instant = Instant::now();
 
+    let mut script_path_blocklist: Vec<String> = Vec::new();
+    loop {
+        // at each iteration, we need to ignore the script paths that have already been
+        // pulled and for which the concurrency limit has been reached. Otherwise each
+        // single worker will loop indefinitely here until the concurrency limit is not
+        // reached anymore
```

Comment:
I don't think that's true since you requeue with a later scheduled_for.

I'm actually not sure why you need to change the pull function at all. Requeue is all that's necessary